test/src/util/key_table_test_suite.c

```diff
@@ -0,0 +1,227 @@
+/*
+ * key_table_test_suite.c
+
+ */
+
+#include "key_table_test_suite.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <CharBuf.h>
+#include <pelz_log.h>
+
+// Adds all key table tests to main test runner.
+int key_table_suite_add_tests(CU_pSuite suite)
+{
+  if(NULL == CU_add_test(suite, "Test Key Table Initialization/Destruction", test_table_initDestroy))
+		  {
+    return (1);
+		  }
+  if(NULL == CU_add_test(suite, "Test all combinations for adding keys to Key Table", test_table_initAddDestroy))
+		  {
+    return (1);
+		  }
+  if(NULL == CU_add_test(suite, "Test all Key Table Lookup combinations", test_table_initLookupAddDestroy))
+		  {
+    return (1);
+		  }
+  if(NULL == CU_add_test(suite, "Test Key Table Delete", test_table_initLookupAddDeleteDestroy))
+		  {
+    return (1);
+		  }
+    return (0);
+}
+
+void test_table_initDestroy(void)
+{
+  CU_ASSERT(key_table_init() == 0)
+  CU_ASSERT(key_table_destroy() == 0)
+}
+
+void test_table_initAddDestroy(void)
+{
+  CharBuf tmp;
+  CharBuf key;
+  char cwd[1024];
+  char *valid_id[3] = {"file:/test/key1.txt", "file:/test/key2.txt", "file:/test/key3.txt"};
+  char *tmp_id;
+
+
+  getcwd(cwd, sizeof(cwd));
+  pelz_log(LOG_DEBUG, "Test Key Table Add Function");
+  CU_ASSERT(key_table_init() == 0)
+
+  //Test that the keys are added to the key table
+  for (int i = 0; i < 3; i++)
+  {
+    tmp = newCharBuf(strlen(valid_id[i]) + strlen(cwd));
```

Comment:
I think a comment here that this is inserting the current working directory into the file path because it's required for `key_table_add` to work properly would be helpful.